aodndata/gsla/handler.py

```diff
@@ -70,74 +71,93 @@ def get_product_type(netcdf_path):
 class GslaHandler(HandlerBase):
     def __init__(self, *args, **kwargs):
         super(GslaHandler, self).__init__(*args, **kwargs)
-        self.allowed_extensions = ['.gz']
+        self.allowed_extensions = ['.gz', '.nc']
 
     def preprocess(self):
-        # add nc_gz file to collection (not by default)
-        self.file_collection.add(self.input_file_object)
-        netcdf_file_gz_collection = self.file_collection.filter_by_attribute_id('file_type', FileType.GZIP)
-        netcdf_file_gz = netcdf_file_gz_collection[0]
-        netcdf_file_gz.publish_type = PipelineFilePublishType.HARVEST_UPLOAD  # default
 
-        # GSLA files are gzipped, so gunzip them before checking them
+        # if input file is a NetCDF, create a .nc.gz and harvest upload it.
+        # historically, files were always sent as *.nc.gz. But as of April 2021, files might be pushed as *.nc.
+        # to be consistent, we transform this .nc into a .nz.gz
+        if self.file_type is FileType.NETCDF:
+            netcdf_collection = self.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)
+            netcdf_file = netcdf_collection[0]
+            netcdf_file.publish_type = PipelineFilePublishType.NO_ACTION
+
+            gzip_path = os.path.join(self.temp_dir,  os.path.basename(self.input_file + '.gz'))
+            with open(self.input_file, 'rb') as f_in, gzip.open(gzip_path, 'wb') as gz_out:
+                gz_out.writelines(f_in)
+
+            # publish
+            netcdf_file_gz = PipelineFile(gzip_path, file_update_callback=self._file_update_callback)
+            netcdf_file_gz.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+
+            self.file_collection.add(netcdf_file_gz)
```

Comment:
And again... (see https://github.com/aodn/python-aodncore/pull/209)
```suggestion
            self.add_to_collection(gzip_path, publish_type=PipelineFilePublishType.HARVEST_UPLOAD)
```